(PLATFORM-3335) Add preference and permission for opting in to HTTPS

Grunny opened the pull request:
Add a user preference to allow users to opt in to HTTPS and a permission
for restricting access to it.

/cc @Wikia/core-platform-team

Grunny committed:
(PLATFORM-3335) Add preference and permission for opting in to HTTPS

Add a user preference to allow users to opt in to HTTPS and a permission
for restricting access to it.

Grunny committed:
(PLATFORM-3336) Handle redirecting users who have opted in to HTTPS

Handle redirecting users who have opted in to HTTPS, and also redirect
users who haven't or are unable to opt in back to HTTP if necessary.

KockaAdmiralac commented:
Since this is a global preference, does the team that handles the user-preference service need to be contacted so they make it check for user's permissions before setting the HTTPS opt-in preference? Otherwise we would easily be able to set the preference by sending a PUT request to https://services.wikia.com/user-preference/{user_id}, bypassing the permission check.

Grunny commented:
@KockaAdmiralac The permissions check here only exposes the preference on the page, but we still check the permission before doing anything: https://github.com/Wikia/app/pull/14535 It will also be only enabled per wiki initially, so it doesn't matter if people who don't have the permission set the preference via the service, as the preference by itself won't have any effect. :)

Grunny committed:
(PLATFORM-3336) Check HTTPS is allowed on the wiki

Check HTTPS is allowed on the wiki as we will now include this extension on
all wikis for when we allow HTTPS requests to hit the backend. Also provide
a variable to allow disabling forcing users back to plain HTTP on dev and
sandbox environments.

Commit:
Merge pull request #14535 from Wikia/PLATFORM-3336

(PLATFORM-3336) Handle redirecting users who have opted in to HTTPS
